sdk/nodejs/cmd/pulumi-language-nodejs/main.go

```diff
@@ -1006,3 +1015,160 @@ func (host *nodeLanguageHost) RunPlugin(
 ) error {
 	return errors.New("not supported")
 }
+
+func (host *nodeLanguageHost) GenerateProject(
+	ctx context.Context, req *pulumirpc.GenerateProjectRequest,
+) (*pulumirpc.GenerateProjectResponse, error) {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return nil, err
+	}
+
+	sink := diag.DefaultSink(os.Stderr, os.Stderr, diag.FormatOptions{
+		Color: cmdutil.GetGlobalColorization(),
+	})
+	pluginCtx, err := plugin.NewContext(sink, sink, nil, nil, cwd, nil, true, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	parser := hclsyntax.NewParser()
+	// Load all .pp files in the directory
```

Comment:
Would you mind elaboring on this? What are .pp files?

Reply:
pcl files, we use the ".pp" suffix for them. If you go looking in the codegen tests you'll see a lot of them.